packages/snaps-utils/src/manifest/validators/unused-exports.ts

```diff
@@ -0,0 +1,80 @@
+import type { InitialPermissions } from '@metamask/snaps-sdk';
+
+import type { ValidatorMeta } from '../validator-types';
+
+/**
+ * Check if all the required files are included.
+ */
+export const unusedExports: ValidatorMeta = {
+  severity: 'warning',
+  semanticCheck(files, context) {
+    const { handlerEndowments, exports } = context.options ?? {};
+
+    // The handler endowments or exports must be provided for this check to be
+    // performed.
+    if (!handlerEndowments || !exports) {
+      return;
+    }
+
+    const handlers = Object.entries(handlerEndowments)
+      .filter(([handler, endowment]) => {
+        if (endowment === null) {
+          return false;
+        }
+
+        return (
+          exports.includes(handler) &&
+          !files.manifest.result.initialPermissions[
+            endowment as keyof InitialPermissions
+          ]
+        );
+      })
+      .map(([handler, endowment]) => `${handler} (${endowment})`);
+
+    const endowments = Object.entries(handlerEndowments).filter(
```

Comment:
Can we consider renaming here? `handler` and `endowments` are used almost interchangeably. Feels like one of these should be `unusedHandlers` and maybe the other `unusedEndowments`?

Reply:
Open to other ways of renaming, just pointing out that it is quite hard to understand as-is

Reply:
I think I misunderstood. Renamed them now!